integration/integration_test.go

```diff
@@ -371,3 +372,51 @@ func TestDemoteDownedManager(t *testing.T) {
 	numManager--
 	pollClusterReady(t, cl, numWorker, numManager)
 }
+
+func TestRestartLeader(t *testing.T) {
+	numWorker, numManager := 10, 5
+	cl := newCluster(t, numWorker, numManager)
+	defer func() {
+		require.NoError(t, cl.Stop())
+	}()
+	leader, err := cl.Leader()
+	require.NoError(t, err)
+
+	nodeID := leader.node.NodeID()
+
+	require.NoError(t, leader.Pause())
+
+	require.NoError(t, raftutils.PollFuncWithTimeout(nil, func() error {
+		resp, err := cl.api.GetNode(context.Background(), &api.GetNodeRequest{
+			NodeID: nodeID,
+		})
+		if err != nil {
+			return err
+		}
+		if resp.Node.Status.State != api.NodeStatus_DOWN {
+			return errors.Errorf("node %s is still not down", nodeID)
+		}
+		return nil
+	}, opsTimeout))
+
+	require.NoError(t, raftutils.PollFuncWithTimeout(nil, func() error {
+		resp, err := cl.api.ListNodes(context.Background(), &api.ListNodesRequest{})
+		if err != nil {
+			return err
+		}
+		for _, node := range resp.Nodes {
+			if node.ID == nodeID {
+				continue
+			}
+			require.False(t, node.Status.State == api.NodeStatus_DOWN, "nodes shouldn't go to down")
+			if node.Status.State != api.NodeStatus_READY {
```

Comment:
It's kinda different :) Not sure how to do it better. It can be also UNKNOWN, which is ok while polling. DOWN shouldn't appear at all.